src/@utils/pool.ts

```diff
@@ -0,0 +1,86 @@
+import { approve, Pool, ZERO_ADDRESS } from '@oceanprotocol/lib'
+import Web3 from 'web3'
+import { getSiteMetadata } from './siteConfig'
+import { getDummyWeb3 } from './web3'
+import { TransactionReceipt } from 'web3-eth'
+
+export async function calculateBuyPrice(
+  accessDetails: AccessDetails,
+  web3?: Web3,
+  chainId?: number
+): Promise<PriceAndEstimation> {
+  if (web3 && chainId)
+    throw new Error("web3 and chainId can't be undefined at the same time!")
```

Comment:
Logic and message are kinda mismatched, what is the intention here?

Reply:
is it clearer now with the description?

Reply:
So the error message should be `web3 and chainId can't be defined at the same time`?

Reply:
nope, undefined at same time, if web3 is there, chainId is ignored .

Reply:
but the logic says, "if `web3` and `chainId` are defined at same time, return an error", that's why I'm a bit confused.

But already shows we should rather deal with that in our web3 context provider and initiate web3 at all times with `config.nodeUri` fallback, not just through our `connect` method cause also doing this in `useGraphSyncStatus`. Then we would not need hacks like this but oh boy, might be fun with web3modal and could become quickly out of scope

Reply:
> but the logic says, "if web3 and chainId are defined at same time, return an error", that's why I'm a bit confused.
my god.... 😆 , it was working properly that's why i didn't notice what i did.

Can't handle that globally because you don't know on what chain you will need it